src/client.js

```diff
@@ -305,427 +305,558 @@ class ConnectionParameters {
+         * @param fileRes
+         * @returns {Promise<void>}
+         * @private
+         */
+    const _publishIfNeeded = async (fileRes) => {
+      await client.NLWS.xtkFileRes.create(fileRes).publishIfNeeded();
+    }
+
+    /**
+         * This Will call the SOAP method(GetURL) which returns the public URL of the uploaded file.
+         * @param fileRes
+         * @returns {Promise<string>}
+         * @private
+         */
+    const _getPublicUrl = async (fileRes) => {
+      return await client.NLWS.xtkFileRes.create(fileRes).getURL()
+    }
+    return {
+      /**
+             * This is the exposed/public method for fileUploader instance which will do all the processing related to the upload process internally and returns the promise containing all the required data.
+             * @param file
+             * @returns {Promise<{name: string, md5: string, type: string, size: string, url: string}>}
+             */
+      upload: (file) => {
+        return new Promise((resolve, reject) => {
+          if (!Util.isBrowser()) {
+            reject('File uploading is only supported in browser based calls.');
+          }
+          try {
+            var data = new FormData()
+            data.append('file_noMd5', file)
+            //TODO: Needs to be refactored after cookie issue get resolved.
+            client._makeHttpCall({
+              url: `${client._connectionParameters._endpoint}/nl/jsp/uploadFile.jsp`,
+              processData: false,
+              credentials: 'include',
+              method: 'POST',
+              body: data,
+              headers: {
+                'x-security-token': client._securityToken,
+                'Cookie': '__sessiontoken=' + client._sessionToken,
+              }
+            }).then((okay)=>{
+              var iframe = document.createElement('iframe');
+              iframe.style.height = 0;
+              iframe.style.width = 0;
+              document.controller = {
+                uploadFileCallBack: async (data) => {
+                  const counter = await _increaseValue(); // Step 1
+                  const fileRes= _createFileRes(counter, data)
+                  await _write(fileRes); // Step 2
+                  await _publishIfNeeded(fileRes) // Step 3
+                  const url = await _getPublicUrl(fileRes) // Step 3
+                  resolve({
+                    name: data[0].fileName,
+                    md5: data[0].md5,
+                    type: file.type,
+                    size: file.size,
+                    url: url
+                  })
+                }
+              }
+              var html = `<body>${okay}</body>`;
+              document.body.appendChild(iframe);
+              iframe.contentWindow.document.open();
+              iframe.contentWindow.document.write(html);
+              iframe.contentWindow.document.close();
+            }).catch((ex)=>{
+              reject(ex);
```

Comment:
All exception types must be consistent (see CampaignException)
The idea is to create an error code and throw a CampaignException with the proper error code (and reuse error codes when possible or when it makes sense)